hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -198,6 +201,28 @@ public void testCreateTableTxnBuilder() throws Exception {
     }
   }
 
+  @Test
+  public void testLoadTableWithCustomMetricReporter() throws Exception {
+    Schema schema = getTestSchema();
+    TableIdentifier tableIdent = TableIdentifier.of(DB_NAME, "tbl");
+    String location = temp.newFolder("tbl").toString();
+
+    HiveCatalog catalog = new HiveCatalog();
+    catalog.initialize(
+        "hive", ImmutableMap.of("metrics-reporter-impl", TestMetricsReporter.class.getName()));
+    try {
+      Transaction txn =
+          catalog.buildTable(tableIdent, schema).withLocation(location).createTransaction();
+      txn.commitTransaction();
+      BaseTable baseTable = (BaseTable) catalog.loadTable(tableIdent);
+      Assertions.assertInstanceOf(TestMetricsReporter.class, baseTable.reporter());
```

Comment:
rather than exposing `reporter()` in `BaseTable`, I think it would be better to actually make sure that the custom reporter's `report()` method is being called

Reply:
sure.

Reply:
I don't see see a way to assert that custom reporter's `report()` method is being called if we don't expose this field. Do we use reflection to get field hidden fields and assert on them?

Reply:
Since, I don't see a way to assert, I have removed this test.